main.go

```diff
@@ -51,7 +60,7 @@ func main() {
 	generateDownloadsProducer, err := kafka.NewProducer(cfg.KafkaAddr, cfg.GenerateDownloadsTopic, 0)
 	if err != nil {
 		log.Error(errors.Wrap(err, "error creating kakfa producer"), nil)
-		os.Exit(1)
+		initialised.generateDownloadsProducer = false
```

Comment:
🤔 I think we need to be careful here as multiple fields could change at the same time which could have weird unintended consequences. 
We can have better control over how/when the field is set if we use a method/function. I would consider adding a lock / `mutex` field to the struct and methods for setting each field. Within each method 
- acquire the lock
- update the field
- release the lock.